### PR TITLE
Make sure the crash reporter doesnt crash itself

### PR DESCRIPTION
**Pull request by ohayon**
Fixes https://github.com/fastlane/fastlane/issues/9197 by wrapping the whole crash reporter execution into a `begin/rescue` block to handle any issue that could come up including not being able to reach servers from certain places in the world.

### CODE REVIEW

**Comment by fastlane-bot-helper** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please provide a changelog summary in the Pull Request description @ohayon</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: fastlane_core/lib/fastlane_core/crash_reporter/crash_reporter.rb [hunk 1]
```diff
@@ -24,12 +24,18 @@ def report_crash(exception: nil, action: nil)
...
-        @did_report_crash = true
+        begin
+          payload = CrashReportGenerator.generate(exception: exception, action: action)
+          send_report(payload: payload)
+          save_file(payload: payload)
+          show_message unless did_show_message?
+          @did_report_crash = true
+        rescue
+          if FastlaneCore::Globals.verbose?
+            UI.error("Unable to send the crash report.")
```
Comment (mpirri) [inline fastlane_core/lib/fastlane_core/crash_reporter/crash_reporter.rb]: Worth logging the exception here?

Comment (ohayon) [inline fastlane_core/lib/fastlane_core/crash_reporter/crash_reporter.rb]: Not sure, do you think there is value in that?

Comment (mpirri) [inline fastlane_core/lib/fastlane_core/crash_reporter/crash_reporter.rb]: Go for it!  At least maybe we have a chance of finding out what is wrong with the crash reporter if we log the error? Downside is that it is a bit wordy and the user has to worry about two crashes at the same time (original error + crash reporter crash), which can be confusing.

**Review by mpirri — APPROVED** [review]
